src/en/troubleshooting.md

```diff
@@ -1,16 +1,310 @@
 
+# Troubleshooting
 
-# Troubleshooting Juju
+Juju does a brilliant job at simplifying the deployment of big software at
+scale. But trying to accommodate every variation of model, cloud and connection
+is a huge challenge, and one that can occasionally cause a few bumps along
+the way to cloud nirvana.
 
-This section is dedicated to the troubleshooting of Juju when things go wrong.
-Make sure you have [Juju installed](./getting-started.html#installation) and
-configured correctly.
+Fortunately, Juju is well equipped to help in instances like these, both in the
+output it can provide and in specific additional steps that can be taken to
+mitigate any problems.
 
-- [Troubleshooting Juju itself](./troubleshooting-juju.html) - `Coming soon!`
-- [Juju logs](./troubleshooting-logs.html)
-- [Troubleshooting model upgrades](./troubleshooting-upgrade.html) - help for
-upgrading Juju across a model.
+## Gathering details
 
-If your issue is not addressed here, consider the
+When things don't seem to be working, the first step should always be to gather
+as much information as possible. Juju offers the following general options for
+data gathering, which we'll cover below:
+
+- [Advanced status output](#advanced-status-outout)
+- [*show* commands](#using-the-show-commands)
+- [Debug details](#using-the---debug-option)
+- [Log files](#collecting-the-logs)
+
+### Advanced status output
+
+The `juju status` command is typically used to show the status of an entire
+model. But it can also be used to target the details for a specific application
+or unit. Typing `juju status mysql`, for example, will show only the status
+information relevant to `mysql`.
+
+The default output from `status` uses a tabular format to fit as much detail as
+possible into a terminal.  However, by specifying `yaml` as an output format,
+additional details are included in its serialised output. 
+
+!!! Note:
+    When filing bugs and requesting help it's almost always better to use the
+    YAML format so that everyone has additional insight into what's going on.
+
+To see all the details on a deployed 'mysql' application, enter the following:
+
+```bash
+juju status mysql --format=yaml
+```
+
+We've split the output from this command into three separate sections so we
+can annotate the most useful parts for troubleshooting.
+
+```yaml
+model:
+  name: documentation-demo
+  controller: jaas
+  cloud: aws
+  region: eu-west-1
+  version: 2.1.2
+```
+
+The above 'model' section includes the region for the deployment and the
+version number of the Juju agent used by the model. 
+
+This version number should ideally match the Juju client version you're using,
+as well as the version used to deploy the controller. 
+
+Type `juju --version` to see the client version and `juju controllers` to see
+the version numbers of any controllers. 
+
+To update a controller and its models, see
+[Upgrading Juju software][modelsupgrade] for further details.
+
+```yaml
```

Comment:
this should still work - i think the exact format of the output has changed though